swift/src/IceImpl/Endpoint.mm

```diff
@@ -64,82 +39,70 @@ + (id)createEndpointInfo:(std::shared_ptr<Ice::EndpointInfo>)infoPtr
+    }
 
-        return [factory createOpaqueEndpointInfo:handle
-                                      underlying:underlying
-                                         timeout:opaqueInfo->timeout
-                                        compress:opaqueInfo->compress
-                                   encodingMajor:opaqueInfo->rawEncoding.major
-                                   encodingMinor:opaqueInfo->rawEncoding.minor
-                                        rawBytes:rawBytes];
+    if (std::dynamic_pointer_cast<Ice::SSL::EndpointInfo>(infoPtr))
+    {
+        return [factory createSSLEndpointInfo:underlying];
     }
 
     auto udpInfo = std::dynamic_pointer_cast<Ice::UDPEndpointInfo>(infoPtr);
     if (udpInfo)
     {
-        return [factory createUDPEndpointInfo:handle
-                                   underlying:underlying
-                                      timeout:udpInfo->timeout
-                                     compress:udpInfo->compress
+        return [factory createUDPEndpointInfo:udpInfo->compress
                                          host:toNSString(udpInfo->host)
                                          port:udpInfo->port
                                 sourceAddress:toNSString(udpInfo->sourceAddress)
                                mcastInterface:toNSString(udpInfo->mcastInterface)
                                      mcastTtl:udpInfo->mcastTtl];
     }
 
-    auto ipInfo = std::dynamic_pointer_cast<Ice::IPEndpointInfo>(infoPtr);
-    if (std::dynamic_pointer_cast<Ice::TCPEndpointInfo>(infoPtr))
-    {
-        return [factory createTCPEndpointInfo:handle
-                                   underlying:underlying
-                                      timeout:ipInfo->timeout
-                                     compress:ipInfo->compress
-                                         host:toNSString(ipInfo->host)
-                                         port:ipInfo->port
-                                sourceAddress:toNSString(ipInfo->sourceAddress)];
-    }
-
     auto wsInfo = std::dynamic_pointer_cast<Ice::WSEndpointInfo>(infoPtr);
     if (wsInfo)
     {
-        return [factory createWSEndpointInfo:handle
-                                  underlying:underlying
-                                     timeout:infoPtr->timeout
-                                    compress:infoPtr->compress
-                                    resource:toNSString(wsInfo->resource)];
+        return [factory createWSEndpointInfo:underlying resource:toNSString(wsInfo->resource)];
     }
 
-    if (std::dynamic_pointer_cast<Ice::SSL::EndpointInfo>(infoPtr))
-    {
-        return [factory createSSLEndpointInfo:handle
-                                   underlying:underlying
-                                      timeout:infoPtr->timeout
-                                     compress:infoPtr->compress];
-    }
-
-#if TARGET_OS_IPHONE
+#if TARGET_OS_IPHONE // TODO: remove conditional
```

Comment:
The IceIAP::EndpointInfo and ConnectionInfo classes are tiny classes that don't need be conditional.